qa/tasks/immutable_object_cache.py

```diff
@@ -0,0 +1,143 @@
+            )
+    try:
+        yield
+    finally:
+        log.info("cleanup immutable object cache")
+        for client, client_config in config.items():
+            client_config = client_config if client_config is not None else dict()
+            (remote,) = ctx.cluster.only(client).remotes.keys()
+            remote.run(
+                args=[
+                    'sudo', 'killall', '-s', '9', 'ceph-immutable-object-cache', run.Raw('||'), 'true',
+                    ]
+                )
+            cache_path = client_config.get('immutable object cache path', '/tmp/ceph-immutable-object-cache')
+            remote.run(
+                args=[
+                    'sudo', 'rm', '-rf', cache_path, run.Raw('||'), 'true',
+                    ]
+                )
+
+@contextlib.contextmanager
+def thrashes_immutable_object_cache_daemon(ctx, config):
+    """
+    thrashes immutable object cache daemon.
+    It can test reconnection feature of RO cache when RO daemon crash
+    TODO : replace sleep with better method.
+    """
+    log.info("thrashes immutable object cache daemon")
+
+    # just thrash one rbd client.
+    client, client_config = list(config.items())[0]
+    (remote,) = ctx.cluster.only(client).remotes.keys()
+    client_config = client_config if client_config is not None else dict()
+    kill_daemon_time = client_config.get('kill_daemon_time', DEFAULT_KILL_DAEMON_TIME)
+    dead_time = client_config.get('dead_time', DEFAULT_DEAD_TIME)
+    live_time = client_config.get('live_time', DEFAULT_LIVE_TIME)
+    wait_for_qemu_boot = client_config.get('qemu_boot_time', DEFAULT_QEMU_BOOT_TIME)
+
+    # wait for all qemu startup
+    log.info("ceph-immutable-object-cache speedup read request.....")
+    remote.run(
+        args=[
+            'sleep', '{wait_for_qemu_boot}'.format(wait_for_qemu_boot=wait_for_qemu_boot),
+            ]
+        )
+
+    for i in range(kill_daemon_time):
+        log.info("ceph-immutable-object-cache crash....")
+        remote.run(
+            args=[
+                'sudo', 'killall', '-s', '9', 'ceph-immutable-object-cache', run.Raw('||'), 'true',
+                 ]
+            )
+        # librbd shoud normally run when ceph-immutable-object-cache
+        remote.run(
+            args=[
+                'sleep', '{dead_time}'.format(dead_time=dead_time),
+                 ]
+            )
+        # librbd should reconnect daemon
+        log.info("startup ceph-immutable-object-cache")
+        remote.run(
+            args=[
+                'ceph-immutable-object-cache', '-b',
+                 ]
+            )
+        remote.run(
+            args=[
+                'sleep', '{live_time}'.format(live_time=live_time),
+                 ]
+            )
+    try:
+        yield
+    finally:
+        log.info("cleanup thrashes immutable object cache")
+
+@contextlib.contextmanager
+def task(ctx, config):
+    """
+    basedun_qemu(ctx=ctx, config=conf on implement of qemu task.
```

Comment:
Nit: lots of `qemu` references

Reply:
> Nit: lots of `qemu` references

sorry,  these are outdated information, I will delete.